xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererGBM.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ *      Copyright (C) 2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "RPRendererGBM.h"
+
+#include "cores/RetroPlayer/process/gbm/RenderBufferGBM.h"
+#include "cores/RetroPlayer/process/gbm/RenderBufferPoolGBM.h"
+#include "cores/RetroPlayer/rendering/RenderContext.h"
+#include "cores/RetroPlayer/rendering/RenderVideoSettings.h"
+#include "utils/GLUtils.h"
+#include "utils/log.h"
+
+#include <cstring>
+#include <stddef.h>
+
+using namespace KODI;
+using namespace RETRO;
+
+// --- CRendererFactoryGBM ------------------------------------------------
+
+std::string CRendererFactoryGBM::RenderSystemName() const
+{
+  return "GBM";
+}
+
+CRPBaseRenderer *CRendererFactoryGBM::CreateRenderer(const CRenderSettings &settings, CRenderContext &context, std::shared_ptr<IRenderBufferPool> bufferPool)
+{
+  return new CRPRendererGBM(settings, context, std::move(bufferPool));
+}
+
+RenderBufferPoolVector CRendererFactoryGBM::CreateBufferPools(CRenderContext &context)
+{
+  return { std::make_shared<CRenderBufferPoolGBM>(context) };
+}
+
+// --- CRPRendererGBM -----------------------------------------------------
+
+CRPRendererGBM::CRPRendererGBM(const CRenderSettings &renderSettings, CRenderContext &context, std::shared_ptr<IRenderBufferPool> bufferPool) :
+  CRPRendererOpenGLES(renderSettings, context, std::move(bufferPool))
+{
+  m_textureTarget = GL_TEXTURE_EXTERNAL_OES;
+}
+
+CRPRendererGBM::~CRPRendererGBM()
+{
+  Deinitialize();
+}
+
+void CRPRendererGBM::Render(uint8_t alpha)
+{
+  CRenderBufferGBM *renderBuffer = static_cast<CRenderBufferGBM*>(m_renderBuffer);
+  assert(renderBuffer);
```

Comment:
I prefer verbosity for additional clarity, e.g. `assert(renderBuffer != nullptr);`